### PR TITLE
Proposed fix for #938

### PR DESCRIPTION
**Pull request by marcelstoer**
Here's the proposed fix for #938. Not being familiar with the rest of the code basis and your coding style made this somewhat tricky. Hope it's close to what you had in mind.

P.S. Sorry that the minor fixes to the README.md show up in this PR again, should have created a separate branch for that.

### CODE REVIEW

File: wro4j-core/src/main/java/ro/isdc/wro/model/factory/XmlModelFactory.java [hunk 1]
```diff
@@ -76,40 +76,40 @@
...
    * false.
    */
-  private static final String ATTR_GROUP_ABSTRACT = "abstract";
+  protected static final String ATTR_GROUP_ABSTRACT = "abstract";
   /**
    * Minimize attribute specified on resource level, used to turn on/off minimization on this particular resource during
    * pre processing.
    */
-  private static final String ATTR_MINIMIZE = "minimize";
+  protected static final String ATTR_MINIMIZE = "minimize";
```
Comment (alexo) [inline wro4j-core/src/main/java/ro/isdc/wro/model/factory/XmlModelFactory.java]: Wondering if it is really necessary to increase visibility of all the constants. Do you see any use-case when the extending class would need to access them?

File: wro4j-core/src/main/java/ro/isdc/wro/model/factory/XmlModelFactory.java [hunk 2]
```diff
@@ -359,17 +372,30 @@ private void parseResource(final Element resourceElement, final Collection<Resou
...
   }
 
   /**
+   * Creates a resource from a given resourceElement. The element is guaranteed to be of a simple non-recursive type
+   * (i.e. css or js).
+   *
+   * @param resourceElement
+   *          Resource element to parse.
+   */
+  protected Resource createResource(final Element resourceElement, final ResourceType type) {
```
Comment (alexo) [inline wro4j-core/src/main/java/ro/isdc/wro/model/factory/XmlModelFactory.java]: I think the ResourceType is something that can be resolved from resourceElement. As result, the ResourceType parameter is redundant.

**Comment by alexo** [comment]
Great, thanks!
Beside a couple of minor changes, everything looks great.

File: wro4j-core/src/main/java/ro/isdc/wro/model/factory/XmlModelFactory.java (hunk 1 above)
Comment (marcelstoer) [inline wro4j-core/src/main/java/ro/isdc/wro/model/factory/XmlModelFactory.java]: If sub classes are to process `Element` instances in `createGroup` and `createResource` they should have access those constants just like the base implementation in `XmlModelFactory` has.

File: wro4j-core/src/main/java/ro/isdc/wro/model/factory/XmlModelFactory.java (hunk 2 above)
Comment (marcelstoer) [inline wro4j-core/src/main/java/ro/isdc/wro/model/factory/XmlModelFactory.java]: Yes, but `parseResource` already determined the type to find out if it's a group reference:

```java
if (TAG_JS.equals(tagName)) {
  type = ResourceType.JS;
} else if (TAG_CSS.equals(tagName)) {
  type = ResourceType.CSS;
} else if (TAG_GROUP_REF.equals(tagName)) {
```

So, if `createResource` is supposed to determine the resource type itself, which would be totally ok for me, then the above block should be changed to:

```java
if (TAG_GROUP_REF.equals(tagName)) {
  // uri in this case is the group name
  resources.addAll(getResourcesForGroup(uri));
}
```

**Pull request merged**